src/main/java/software/amazon/awssdk/crt/s3/S3ResponseContext.java

```diff
@@ -0,0 +1,33 @@
+package software.amazon.awssdk.crt.s3;
+
+public class S3ResponseContext {
+    private final int errorCode;
+    private final int responseStatus;
+    private final byte[] errorPayload;
+    private final ChecksumAlgorithm checksumAlgorithm;
+    private final boolean didValidateChecksum;
+
+    public S3ResponseContext(final int errorCode, final int responseStatus, final byte[] errorPayload, final ChecksumAlgorithm checksumAlgorithm, final boolean didValidateChecksum) {
+        this.errorCode = errorCode;
+        this.responseStatus = responseStatus;
+        this.errorPayload = errorPayload;
+        this.checksumAlgorithm = checksumAlgorithm;
+        this.didValidateChecksum = didValidateChecksum;
+    }
+
+    public int getErrorCode () {
+        return this.errorCode;
+    }
+    public int getResponseStatus () {
+        return this.responseStatus;
```

Comment:
It's confusing what this value will be if the operation fails (say, due to connection failures where no actual responses ever came back)

Will it be -1? 0?
Is this something the user should only query when the exception is non-null?

Reply:
Looking through the code I believe the callback isn't called in the event a connection timeout, but not positive.

Reply:
I ran the `s_test_s3_put_single_part_fail_object_inputstream_fail_reading` test in aws-c-s3 through the debugger, and the response status is left as 0 for this kind of failure.